FormidableLabs/victory-docs
# Refactor sidebar components

My goal here was to make this a little easier to read and update to use modern React structure w/ hooks and components. 
- Updated partials from class components to functional 
- remove unnecessary brackets on simple components 
- delete out of date notes 
- move early returns to top of components instead of as defaults in ternaries
- refactor table of contents (this is the big chunk) - separating out functionality where i could and updating how the various subsections of the sidebar are getting assembled so that we only have to go to that function once and create an object instead of individual arrays.  This section was hard for me to read while updating the styles last month so I'm hoping now it's slightly easier to follow? The visibility of subsections + the data structure make me hesitant to lean too far into a refactor all at once on this. 

To test:
- sidebar should function as it previously did (all sections at the top should start collapsed, when a section is selected (besides the intro section) it should expand, the lead section title that is active should show that gray background. 
- please also try building this branch locally. i was battling some weird build error that looks like `Failed exporting HTML for URL gallery/area-hover-styles (/Users/brittanyfee  nstra/repos/victory-docs/src/pages/gallery-item-template): Cannot read property 'filte  r' of undefined`  which seems to have disappeared of its own accord but also i didn't touch anything to do with the gallery? 👀 


If you see anything that should be included here let me know, i think i stared at the table of contents stuff too long.
